### PR TITLE
Don't repeatedly mark queued tasks as ready to run.

### LINKED ISSUE #1787: High CPU usage with queued tasks
We still see high CPU usage in large suites when there are queued tasks, though the problem isn't as bad since the fix for #1744.

The 1744 fix stopped the state files being written every second, but in scheduler.py
```
                self.pool.match_dependencies()
```
is still getting called every second (if there are queued tasks) rather than just whenever the state changes. With a lot of waiting tasks this can be expensive.

Example cylc scan output from a large NWP ensemble suite (u-aa670)

```
Task state totals:
      queued:30 runahead:109 running:1 submitted:9 succeeded:479 waiting:249
      20150626T1800Z succeeded:9
      20150627T0000Z queued:30 running:1 submitted:7 succeeded:235 waiting:51
      20150627T0600Z succeeded:140 waiting:147
      20150627T1200Z submitted:2 succeeded:95 waiting:51
      20150627T1800Z runahead:105
      20150628T0000Z runahead:1
      20150628T0600Z runahead:3
```

When there are queued tasks CPU usage is about 30%, otherwise it's about 2%. Running a profiler shows that most of this is from ```prerequisite.satisfy_me``` which is called an enormous number of times.

### PR DESCRIPTION
**Pull request by hjoliver**
Close #1787 

New logic explained by the new in-source comment.

Test battery passes here.

@matthewrmshin - please review/assign

(I can probably make a new test for this, that relies on counting logged iterations through the main scheduling loop in a suite with two tasks - one queued task and one running ... but that'll have to be tomorrow).

### CODE REVIEW

**Comment by matthewrmshin** [comment]
I guess this was missed by #1745.

Looks OK. Test battery passes. I cannot think of a good way to add an automated test for this, however.

@benfitzpatrick please review 2.

File: tests/lib/bash/test_header [hunk 1]
```diff
@@ -235,6 +237,22 @@ function grep_ok() {
     fail $TEST_NAME
 }
 
+function count_ok() {
+    local BRE=$1
+    local FILE=$2
+    local COUNT=$3
+    local TEST_NAME=$(basename "$FILE")-count-ok
+    NEW_COUNT=$(grep -e "$BRE" "$FILE" | wc -l)
```
Comment (benfitzpatrick) [inline tests/lib/bash/test_header]: Could use `grep -c` to count it?

**Comment by benfitzpatrick** [comment]
One comment, otherwise good to go.

**Comment by hjoliver** [comment]
I've made the one small suggested change and verified it's correctness.  I'll take "otherwise good to go" as license to merge now.

**Pull request merged**